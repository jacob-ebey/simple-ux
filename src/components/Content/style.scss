@import '../variables';

.content {
  max-width: $content-width;
  margin: 0 auto;
  padding: $content-padding;
}