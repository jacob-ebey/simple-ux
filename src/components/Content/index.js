import { h } from 'preact' // eslint-disable-line no-unused-vars

import { merge } from '../tools'

import './style.scss'

const Content = ({ class: className, children, ...rest }) => (
  <div
    {...rest}
    class={merge('content', className)}
  >
    {children}
  </div>
)

export default Content
