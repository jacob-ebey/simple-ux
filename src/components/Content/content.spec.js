/* eslint-env jest */
import { h } from 'preact' // eslint-disable-line no-unused-vars
import { shallow } from 'preact-render-spy'

import Content from './'

describe('Content', () => {
  describe('- snapshots -', () => {
    test('can render', () => {
      const wrapper = shallow(<Content />)
      expect(wrapper).toMatchSnapshot()
    })

    it('can add custom class', () => {
      const wrapper = shallow(<Content class='custom' />)
      expect(wrapper).toMatchSnapshot()
    })

    test('can render children', () => {
      const wrapper = shallow(
        <Content>
          <p>Child 1</p>
          <p>Child 2</p>
        </Content>
      )
      expect(wrapper).toMatchSnapshot()
    })
  })
})
