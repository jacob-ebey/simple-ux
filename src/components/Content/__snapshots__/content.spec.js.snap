// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content - snapshots - can add custom class 1`] = `
preact-render-spy (1 nodes)
-------
<div class="content custom"></div>

`;

exports[`Content - snapshots - can render 1`] = `
preact-render-spy (1 nodes)
-------
<div class="content"></div>

`;

exports[`Content - snapshots - can render children 1`] = `
preact-render-spy (1 nodes)
-------
<div class="content">
  <p>Child 1</p>
  <p>Child 2</p>
</div>

`;
